pkg/api/service.go

```diff
@@ -24,19 +27,22 @@ const (
 type Service struct {
 	message_api.UnimplementedReplicationApiServer
 
-	ctx             context.Context
-	log             *zap.Logger
-	registrant      *registrant.Registrant
-	store           *sql.DB
-	publishWorker   *publishWorker
-	subscribeWorker *subscribeWorker
+	ctx              context.Context
+	log              *zap.Logger
+	registrant       *registrant.Registrant
+	store            *sql.DB
+	publishWorker    *publishWorker
+	subscribeWorker  *subscribeWorker
+	messagePublisher blockchain.IMessagePublisher
```

Comment:
Could we name this something like `blockchainPublisher` to be clear, here and in `server.go`? There are multiple places publishes can go, including into the DB.